SetupDataPkg/Library/ConfigKnobShimLib/ConfigKnobShimDxeLib/GoogleTest/ConfigKnobShimDxeLibGoogleTest.cpp

```diff
@@ -0,0 +1,177 @@
+/** @file ConfigKnobShimDxeLibGoogleTest.cpp
+  This is a sample to demonstrates the use of GoogleTest that supports host
+  execution environments.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <GoogleTest/Library/MockUefiRuntimeServicesTableLib.h>
+extern "C" {
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+
+#include "../../ConfigKnobShimLibCommon.c" // include the c file to be able to unit test static function

```

Comment:
could the source file be included in the INF file's section? 
Then just add the C file declarations here?